h2o-core/src/main/java/water/FailedNodeWatchdogExtension.java

```diff
@@ -128,7 +130,12 @@ public void run() {
             try {
                 sleep(watchdogClientConnectTimeout);
                 boolean watchDogConnected = false;
-                for(H2ONode client: H2O.getClients()){
+                HashSet<H2ONode> clients = H2O.getClients();
```

Comment:
A good practise is to check if debug is enabled before entering a complex code used only for debugging

Reply:
Good point, thanks!